webapp/channels/src/components/mfa/confirm.tsx

```diff
@@ -30,57 +30,54 @@ type Props = {
     updateParent: (state: MFAControllerState) => void;
 }
 
-export default class Confirm extends React.PureComponent<Props> {
-    public componentDidMount(): void {
-        document.body.addEventListener('keydown', this.onKeyPress);
-    }
-
-    public componentWillUnmount(): void {
-        document.body.removeEventListener('keydown', this.onKeyPress);
-    }
-
-    submit = (e: KeyboardEvent | React.FormEvent<HTMLFormElement>): void => {
+export default function Confirm({state, updateParent}: Props) {
+    const submit = (e: KeyboardEvent | React.FormEvent<HTMLFormElement>): void => {
         e.preventDefault();
         redirectUserToDefaultTeam();
     };
 
-    onKeyPress = (e: KeyboardEvent | React.FormEvent<HTMLFormElement>): void => {
-        if (isKeyPressed(e as KeyboardEvent, KeyCodes.ENTER)) {
-            this.submit(e);
+    const onKeyPress = useCallback((e: KeyboardEvent) => {
+        if (isKeyPressed(e, KeyCodes.ENTER)) {
+            submit(e);
         }
-    };
+    }, [submit]);
```

Comment:
Once submit is outside of the component, this also doesn't depend on anything inside the component, so it can be moved outside.

Also, just as a note, the way that it is right now implemented, the useCallback is rendered useless. `submit` is not being memoized, so it is being created on every render. Therefore, the dependencies for onKeyPress will always be different, and will re-generate onKeyPress on every render.